File: patterns/java/src/refactoring_guru/patterns/observer/example/EmailNotificationListener.java
package refactoring_guru.patterns.observer.example;

import java.io.File;
import java.io.IOException;

public class EmailNotificationListener implements EventListener{
    private String email;

    public EmailNotificationListener(String email) {
        this.email = email;
    }

    @Override
    public void update(File file) {
        try {
            System.out.println(email + ", " + "Someone has changed the file: " + file.getCanonicalPath());
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }
}

File: patterns/cs/src/RefactoringGuru/Observer/Example/Listeners/EmailNotificationListener.cs
using System;
using System.IO;

namespace RefactoringGuru.Observer.Example.Listeners
{
    class EmailNotificationListener : EventListener
    {
        private String email;

        public EmailNotificationListener(String email)
        {
            this.email = email;
        }
        public void Update(string eventType, FileInfo file)
        {
            Console.WriteLine("Email to " + email + ": Someone has performed " + eventType + " operation with the following file: " + file.Name);
        }
    }
}
